reportportal_client/service_async.py

```diff
@@ -147,24 +148,26 @@ def terminate(self, nowait=False):
             nowait: set to True to terminate immediately and skip processing
                 messages still in the queue
         """
-        logger.debug("Terminating service")
+        with self.lock:
```

Comment:
I would add one more log message above the locking like:
```
log.debug("Acquiring terminating lock")
with self.lock:
    logger.debug("Terminating service")
```

This will allow to better trace status of treads.